src/analysis/modals/ComputeModal/AzureComputeModal/AzureComputeModal.test.ts

```diff
@@ -41,24 +51,24 @@ const persistentDiskModalProps = {
   tool: runtimeToolLabels.JupyterLab,
   workspace: defaultAzureWorkspace,
   location: testAzureDefaultRegion,
+  isLoadingCloudEnvironments: false,
 };
 
-const defaultAjaxImpl = {
-  Runtimes: {
-    runtime: () => ({
-      details: jest.fn(),
-    }),
-  },
-  Buckets: { getObjectPreview: () => Promise.resolve({ json: () => Promise.resolve(imageDocs) }) },
+type AjaxContract = ReturnType<typeof Ajax>;
+
+const defaultAjaxImpl: AjaxContract = {
+  Buckets: { getObjectPreview: () => Promise.resolve({ json: () => Promise.resolve(imageDocs) }) } as Partial<
+    AjaxContract['Buckets']
+  >,
   Disks: {
     disk: () => ({
       details: jest.fn(),
     }),
-  },
+  } as Partial<AjaxContract['Disks']>,
   Metrics: {
-    captureEvent: () => jest.fn(),
-  },
-};
+    captureEvent: jest.fn(),
+  } as Partial<AjaxContract['Metrics']>,
+} as Partial<AjaxContract> as AjaxContract;
```

Comment:
switching to:

`satisfies Partial<AjaxContract> as AjaxContract`

and

`satisfies Partial<AjaxContract['SubContract']
`

should give you improved type safety....` as` works but can be a bit more wiggly then desired.

Reply:
Hmm... I was able to do this for a few part but it is not super straight forward. Since the tests are passing and the next PR will do away with a good bit of the need for this mocking, I'll punt on modifying these 1.5k lines of code to conform to what should have been done when they were written.

Reply:
yeah, I was kinda riding that trade-off too, but yours is a good plan for any easy-to-slip-in improvement.